UE/Application/Ports/BtsPort.cpp

```diff
@@ -1,146 +1,172 @@
-      phoneNumber(phoneNumber)
-{}
-
-void BtsPort::start(IBtsEventsHandler &handler)
-{
-    transport.registerMessageCallback([this](BinaryMessage msg) {handleMessage(msg);});
-    transport.registerDisconnectedCallback([this]{handleDisconnected();});
-    this->handler = &handler;
-}
+    BtsPort::BtsPort(common::ILogger &logger, common::ITransport &transport, common::PhoneNumber phoneNumber)
+        : logger(logger, "[BTS-PORT]"),
+          transport(transport),
+          phoneNumber(phoneNumber)
+    {
+    }
 
-void BtsPort::stop()
-{
-    transport.registerMessageCallback(nullptr);
-    transport.registerDisconnectedCallback(nullptr);
-    handler = nullptr;
-}
+    void BtsPort::start(IBtsEventsHandler &handler)
+    {
+        transport.registerMessageCallback([this](BinaryMessage msg) { handleMessage(msg); });
+        transport.registerDisconnectedCallback([this] { handleDisconnected(); });
+        this->handler = &handler;
+    }
 
-void BtsPort::handleDisconnected()
-{
-    logger.logError("disconnected");
-    handler->handleDisconnected();
-}
+    void BtsPort::stop()
+    {
+        transport.registerMessageCallback(nullptr);
+        transport.registerDisconnectedCallback(nullptr);
+        handler = nullptr;
+    }
 
-void BtsPort::handleMessage(BinaryMessage msg)
-{
-    try
+    void BtsPort::handleDisconnected()
     {
-        common::IncomingMessage reader{msg};
-        auto msgId = reader.readMessageId();
-        common::PhoneNumber from = reader.readPhoneNumber();
-        auto to = reader.readPhoneNumber();
+        logger.logError("disconnected");
+        handler->handleDisconnected();
+    }
 
-        switch (msgId)
-        {
-        case common::MessageId::Sib:
-        {
-            auto btsId = reader.readBtsId();
-            handler->handleSib(btsId);
-            break;
-        }
-        case common::MessageId::AttachResponse:
-        {
-            bool accept = reader.readNumber<std::uint8_t>() != 0u;
-            if (accept)
-                handler->handleAttachAccept();
-            else
-                handler->handleAttachReject();
-            break;
-        }
-        case common::MessageId::Sms:
-        {
-            std::string message = reader.readRemainingText();
-            logger.logDebug("BtsPort, SmsRecieved from: ", from);
-            logger.logDebug("BtsPort, SmsRecieved msg: ", message);
-            handler->handleSmsReceived(from, message);
-            break;
-        }
-        case common::MessageId::CallRequest:
-        {
-            logger.logDebug("BtsPort, Call request from:", from);
-            handler->handleCallRequest(from);
-            break;
-        }
-        case common::MessageId::CallAccepted:
-        {
-            logger.logDebug("BtsPort, Call accepted from", from);
-            handler->handleCallAccepted(from);
-            break;
-        }
-        case common::MessageId::CallTalk:
+    void BtsPort::handleMessage(BinaryMessage msg)
```

Comment:
Tu chyba błąd, w sensie BinraryMessage jest w common::BinraryMessage i nie widzę, żebyś gdzieś to importował lub usage zrobił.

Reply:
Binary Message jest importowane przez inny import xD